src/material/badge/badge.ts

```diff
@@ -238,15 +254,37 @@ export class MatBadge extends _MatBadgeBase implements OnInit, OnDestroy, CanDis
       this._badgeElement.textContent = newContentNormalized;
     }
 
+    if (!this._isHostInteractive()) {
```

Comment:
This fires in `ngOnInit` and whenever the content changes. Isn't it possible for the element to become non-interactive between those two, e.g. if it's on a button that becomes disabled.